tests/test_iqm_backend.py

```diff
@@ -61,6 +61,24 @@ def test_max_circuits(backend):
     assert backend.max_circuits is None
 
 
+def test_qubit_name_to_index_to_qubit_name(adonis_architecture_shuffled_names):
+    client = mock(IQMClient)
+    when(client).get_quantum_architecture().thenReturn(adonis_architecture_shuffled_names)
+    backend = IQMBackend(client)
+
+    assert backend.qubit_name_to_index('QB1') == 0
+    assert backend.qubit_name_to_index('QB2') == 1
+    assert backend.qubit_name_to_index('QB3') == 2
+    assert backend.qubit_name_to_index('QB4') == 3
+    assert backend.qubit_name_to_index('QB5') == 4
```

Comment:
To avoid the struggle when Apollo is needed :-)

```suggestion
    indices = list(range(0, 5))
    indices_from_qubits = list(backend.qubit_name_to_index(i) for i in indices)
    assert indices == indices_from_qubits
```

---

Alternatively, there is unittest's [assertSequenceEqual](https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertSequenceEqual), for which we can simplify:

```python
    indices = range(0, 5)
    indices_from_qubits = (backend.qubit_name_to_index(i) for i in indices)
    assertSequenceEqual(indices, indices_from_qubits)
```